collector/generic_collector.go

```diff
@@ -45,8 +57,20 @@ func NewCollector(collectorName string, configfile string) (*GenericCollector, e
 
 	//TODO : Add checks for validity of config file (eg : Accurate JSON fields)
 
+	minPollFrequency := configInJSON.MetricsConfig[0].PollingFrequency
```

Comment:
We should check that there is at least one (we have the TODO above, but we should at least check that so we don't crash)